File: masar_ksa_hrms/custom/leave_application/leave_application.py
import frappe 


def on_submit(self , method):
    empty_short_leave_application_validation(self)
    
def empty_short_leave_application_validation(self): 
    if self.custom_sla_reference is not None : 
        frappe.throw(f'The Short Leave Application cannot contain any entries.' , title=frappe._('Validation Error'))

File: masar_ksa_hrms/custom/leave_application/leave_application.js


frappe.ui.form.on('Leave Application' , {
    refresh: function(frm){
        ResetReferance(frm);
    }, 
    onload: function(frm){
        ResetReferance(frm);
    }, 
    setup: function(frm){
        ResetReferance(frm);
    }, 
});


function ResetReferance(frm){
    if (frm.doc.docstatus === 0 ){
        frm.doc.custom_sla_reference = null; 
        refresh_field("custom_sla_reference");
        }
}
